config/config.go

```diff
@@ -703,10 +705,29 @@ type QueueConfig struct {
 	// Maximum time sample will wait in buffer.
 	BatchSendDeadline model.Duration `yaml:"batch_send_deadline,omitempty"`
 
+	// Configuration for retrying functionality in remote-write component.
+	Retry Retry `yaml:"retry,omitempty"`
+}
+
+type Retry struct {
 	// On recoverable errors, backoff exponentially.
-	MinBackoff       model.Duration `yaml:"min_backoff,omitempty"`
-	MaxBackoff       model.Duration `yaml:"max_backoff,omitempty"`
-	RetryOnRateLimit bool           `yaml:"retry_on_http_429,omitempty"`
+	MinBackoff  model.Duration `yaml:"min_backoff,omitempty"`
+	MaxBackoff  model.Duration `yaml:"max_backoff,omitempty"`
+	OnRateLimit bool           `yaml:"on_http_429,omitempty"`
+	Policy      RetryPolicy    `yaml:"policy,omitempty"`
+}
+
+// RetryPolicy configures the retry functionality of remote-write component. If unset, the default behaviour is to
+// retry forever.
+type RetryPolicy struct {
+	MinSampleAge model.Duration `yaml:"min_sample_age,omitempty"`
+	MaxRetries   int            `yaml:"max_retries,omitempty"`
+}
+
+// IsDefault returns true if the configuration in RetryPolicy is unset.
+func (p RetryPolicy) IsSet() bool {
+	var tmp RetryPolicy
+	return !(p.MaxRetries == tmp.MaxRetries && p.MinSampleAge == tmp.MinSampleAge)
```

Comment:
where is `tmp` defined?

Reply:
`tmp` is just used here to refer to the default values, so that it's easier to read. But changing `RetryPolicy` makes the job easier and this is no more required. Thank you for the suggestion.